src/app/shared/components/template/components/task-progress-bar/task-progress-bar.component.ts

```diff
@@ -1,81 +1,175 @@
-import { Component, EventEmitter, Input, OnInit, Output } from "@angular/core";
+import {
+  ChangeDetectionStrategy,
+  ChangeDetectorRef,
+  Component,
+  EventEmitter,
+  Input,
+  OnInit,
+  Output,
+} from "@angular/core";
 import { TaskService } from "src/app/shared/services/task/task.service";
-import { getStringParamFromTemplateRow } from "src/app/shared/utils";
+import {
+  getBooleanParamFromTemplateRow,
+  getStringParamFromTemplateRow,
+} from "src/app/shared/utils";
 import { TemplateBaseComponent } from "../base";
 import { TemplateFieldService } from "../../services/template-field.service";
 import { AppDataService } from "src/app/shared/services/data/app-data.service";
 import { IProgressStatus } from "src/app/shared/services/task/task.service";
 import { CampaignService } from "src/app/feature/campaign/campaign.service";
+import { Subscription, debounceTime } from "rxjs";
+import { DynamicDataService } from "src/app/shared/services/dynamic-data/dynamic-data.service";
+
+interface ITaskProgressBarParams {
+  /**
+   * TEMPLATE PARAMETER: task_group_data.
+   * The name of the task group to track the progress of (through the completion of its subtasks)
+   */
+  taskGroupDataList: string;
+  /**
+   * TEMPLATE PARAMETER: completed_field.
+   * The name of the field that stores the completion status of the task group
+   */
+  taskGroupCompletedField: string;
+  /**
+   * TEMPLATE PARAMETER: progress_units_name.
+   * The displayed name of the units of progress associated with
+   * the constituent tasks of the task group.
+   * Default "sections"
+   */
+  progressUnitsName: string;
+  /**
+   * TEMPLATE PARAMETER: show_text.
+   * Show the text associated with progress bar, e.g. "8 sections"
+   * Default false
+   */
+  showText: boolean;
+  /**
+   * TEMPLATE PARAMETER: variant.
+   * A list of named style variants of the component, separated by spaces or commas.
+   * */
+  variant: "green" | "secondary";
+  /**
+   * TEMPLATE PARAMETER: dynamic.
+   * Specify whether to use the dynamic data feature:
+   * Should be true if the task_group_data has a "completed" column,
+   * else should be false if the task_group_data has a "completed_field" column
+   * */
+  useDynamicData: boolean;
```

Comment:
nit(non-blocking) - would it be confusing for authors to interpret what `dynamic` should be? I'm guessing this should actually be the default behaviour if available. Would it be better here for authors to just pass a `completed_column_name` variable (which defaults to `'completed'`) and look to deprecate the `completed_field` syntax?

Alternatively we could add a means to subscribe to field value changes to keep dynamic behaviour

Reply:
Good point, I don't think `dynamic` would make any sense to authors. I've opted to replace with a param called `completed_column_name`, with default value "completed", and a check to see whether the named column is present in the source data list. If the column is not present, the component will look for a column named "completed_field" and will update non-dynamically